### PR TITLE
vmui: add trace analyzer

### PR DESCRIPTION
**Pull request by Loori-R**
Add a tab where the user can paste JSON or upload files with trace information.
When files are uploaded successfully or pasted JSON, all trace information shows for the user.

#3308

### CODE REVIEW

File: app/vmui/packages/vmui/src/pages/TracePage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        <Box
+          display={"flex"}
+          gap={1}
+          flexDirection={"column"}
+          alignItems={"center"}
+          justifyContent={"center"}
+          flexGrow={"1"}
+          paddingBottom={"64px"}
+        >
+          <div>You can choose a file and download it.</div>
```
Comment (zekker6) [inline app/vmui/packages/vmui/src/pages/TracePage/index.tsx:105]:
```suggestion
          <div>Please, file with JSON response content.</div>
```

File: app/vmui/packages/vmui/src/pages/TracePage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+          <div>The file must contain tracing information in JSON format.</div>
```
Comment (zekker6) [inline app/vmui/packages/vmui/src/pages/TracePage/index.tsx:106]:
```suggestion
          <div>The file must contain tracing information in JSON format. In order to use tracing ploease refer to the doc: https://docs.victoriametrics.com/#query-tracing</div>
```

**Review by zekker6 — COMMENTED** [review]
LGTM!

Left a few comments regarding text used for tooltip.

File: app/vmui/packages/vmui/src/pages/TracePage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  const handleOnload = (e: ProgressEvent<FileReader>, filename: string) => {
+    try {
+      const result = String(e.target?.result);
+      const resp = JSON.parse(result);
+      const traceData = resp.trace || resp;
+      if (!traceData.duration_msec) {
+        handleError(new Error("Not found the tracing information"), filename);
+        return;
+      }
+      setTracesState(prev => [...prev, new Trace(traceData, filename)]);
```
[the remark this answers is not in the document]
Comment (Loori-R) [reply app/vmui/packages/vmui/src/pages/TracePage/index.tsx:27]: `new Trace` will not return an error. For example:

File: app/vmui/packages/vmui/src/pages/TracePage/index.tsx (hunk 2 above)
Comment (Loori-R) [reply app/vmui/packages/vmui/src/pages/TracePage/index.tsx]: added link d1d4f89e8dcd932b06916aa519b47b91d95cff7a

File: app/vmui/packages/vmui/src/pages/TracePage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        <Box
+          display={"flex"}
+          flexDirection={"column"}
+          alignItems={"center"}
+          justifyContent={"center"}
+          flexGrow={"1"}
+          paddingBottom={"64px"}
+        >
+          <Typography sx={{fontSize: 16, whiteSpace: "pre-line", textAlign: "center", lineHeight: "26px"}}>
+            Please, file with JSON response content.
```
Comment (zekker6) [inline app/vmui/packages/vmui/src/pages/TracePage/index.tsx:106]:
```suggestion
            Please, upload file with JSON response content.
```

**Review by dmitryk-dk — APPROVED** [review]
LGTM!

**Review by zekker6 — APPROVED** [review]

**Comment by valyala** [comment]
Please see my comments at https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3308#issuecomment-1308533158

**Comment by valyala** [comment]
@Loori-R , please resolve merge conflicts and also take a look at my comments at https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3308#issuecomment-1308533158 . It is better from usability point of view to provide an input textfield for copy-n-pasting the JSON trace instead of the functionality for file upload.

**Review by dmitryk-dk — APPROVED** [review]
LGTM!

**Comment by Loori-R** [comment]
@valyala Added support copy-n-pasting the JSON

**Comment by valyala** [comment]
@Loori-R , thanks for the query trace analyzer feature!

**Comment by valyala** [comment]
FYI, this pull request has been included in [VictoriaMetrics v1.84.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

**Pull request merged**